cmd/cloudinit/hostname/hostname.go

```diff
@@ -0,0 +1,14 @@
+package hostname
+
+func SetHostname(hostname string) error {
+	osType := getOsType()
+
+	if osType == "busybox" {
+		return bb_setHostname(hostname)
+	}
+	return bb_setHostname(hostname)
+}
+
+func getOsType() string {
```

Comment:
Your already using this pattern in https://github.com/rancherio/os/pull/109/files.  Can we not repeat code somehow?